drivers/serial/uart_ns16550.c

```diff
@@ -1425,8 +1425,7 @@ static const struct uart_driver_api uart_ns16550_driver_api = {
 	};                                                                           \
 	DEVICE_DT_INST_DEFINE(n, &uart_ns16550_init, NULL,                           \
 			      &uart_ns16550_dev_data_##n, &uart_ns16550_dev_cfg_##n, \
-			      COND_CODE_1(CONFIG_UART_NS16550_PARENT_INIT_LEVEL,     \
```

Comment:
CONFIG_UART_NS16550_PARENT_INIT_LEVEL was introduced to support UART driver in PCI where PCI bus driver use ACPI to get platform interrupt routing table (ACPI currently support only post kernel). So this change will break few Intel platform currently supporting

Reply:
the change introducing this Kconfig and the change in init levels already broke few platforms. This fixes the issue, so please verify before assuming it will break anything else and please confirm that the tests exposing this issue work for you as well without this change.

Reply:
Can we move ACPI init earlier?

Reply:
probably, where is started now?

Reply:
before:

```
EARLY
PRE_KERNEL_1
  __init_efi_console_init: efi_console_init(NULL)
  __init_statics_init_pre: statics_init(NULL)
  __init___device_dts_ord_3: loapic_init(__device_dts_ord_3)
  __init___device_dts_ord_8: ioapic_init(__device_dts_ord_8)
  __init_pcie_init: pcie_init(NULL)
  __init___device_dts_ord_55: uart_ns16550_init(__device_dts_ord_55)
  __init_uart_console_init: uart_console_init(NULL)
PRE_KERNEL_2
  __init_sys_clock_driver_init: sys_clock_driver_init(NULL)
POST_KERNEL
  __init_enable_logger: enable_logger(NULL)
  __init_malloc_prepare: malloc_prepare(NULL)
  __init___device_dts_ord_32: uart_ns16550_init(__device_dts_ord_32)
  __init___device_dts_ord_33: uart_ns16550_init(__device_dts_ord_33)
APPLICATION
SMP
```

after:

```
EARLY
  __init_pcie_init: pcie_init(NULL)
PRE_KERNEL_1
  __init_efi_console_init: efi_console_init(NULL)
  __init_statics_init_pre: statics_init(NULL)
  __init___device_dts_ord_3: loapic_init(__device_dts_ord_3)
  __init___device_dts_ord_8: ioapic_init(__device_dts_ord_8)
  __init___device_dts_ord_32: uart_ns16550_init(__device_dts_ord_32)
  __init___device_dts_ord_33: uart_ns16550_init(__device_dts_ord_33)
  __init___device_dts_ord_55: uart_ns16550_init(__device_dts_ord_55)
  __init_uart_console_init: uart_console_init(NULL)
PRE_KERNEL_2
  __init_sys_clock_driver_init: sys_clock_driver_init(NULL)
POST_KERNEL
  __init_enable_logger: enable_logger(NULL)
  __init_malloc_prepare: malloc_prepare(NULL)
APPLICATION
SMP
```